### PR TITLE
Add Swift Salamander extension

### PR DESCRIPTION
**Pull request by c0desurfer**
## Description

Adds the Swift Salamander extension for macOS. It hands local navigation requests to the installed dual-pane file manager through the versioned `swiftsalamander://v1` URL protocol.

The default workflow stays deliberately small:

- **Open Path** accepts one value, including `~`, `~/Downloads`, absolute paths, and local `file:` URLs.
- **Open Clipboard Path** and **Open Finder Selection** have no arguments.
- The extension preference **Open Paths In** chooses Active Pane, Opposite Pane, or New Tab for every path command.
- Folder Pair and Saved Workspace are advanced commands and are disabled by default.

The extension does not read file contents, persist paths, run a background process, or send telemetry. It requires a Swift Salamander build that supports the public URL protocol.

## Validation

- `npm test` — 7 tests passed
- `npm run lint` — passed
- `npm run build` — distribution build passed
- Development extension exercised with the packaged Swift Salamander URL handler, including `~` resolving to the user's home directory

## Screencast

The primary commands are no-view URL handoffs. Reviewers can run **Open Path** with `~` and change its destination under Raycast Settings → Extensions → Swift Salamander. **Open Folder Pair** provides the only form view and starts disabled.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and tested the extension in Raycast
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/swift-salamander/src/salamander.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { normalizeLocalPath } from "./path-input";
+
+export type Target = "active" | "opposite" | "new-tab";
+
+type ExtensionPreferences = { pathTarget: Target };
+
+export { normalizeLocalPath } from "./path-input";
+
+export function preferredPathTarget(): Target {
+  return getPreferenceValues<ExtensionPreferences>().pathTarget;
```
[the remark this answers is not in the document]
Comment (c0desurfer) [reply extensions/swift-salamander/src/salamander.ts]: Fixed in c9cdcf58. Preferences and command arguments now use the types generated from the Raycast manifest: Preferences, Arguments.OpenPath, and Arguments.OpenWorkspace. A regression test prevents handwritten manifest types from returning.

File: extensions/swift-salamander/package.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      "name": "open-finder-selection",
+      "title": "Open Finder Selection",
+      "description": "Reveal Finder's selected items in Swift Salamander",
+      "mode": "no-view"
+    },
+    {
+      "name": "open-folder-pair",
+      "title": "Open Folder Pair",
+      "description": "Open two directories together in a new dual-pane tab",
+      "mode": "view",
```
[the remark this answers is not in the document]
Comment (c0desurfer) [reply extensions/swift-salamander/package.json]: Fixed in c9cdcf58. I removed the disabled Open Folder Pair form from the initial release, so every shipped command is now no-view and no screenshot metadata is required. The command can return later with proper Raycast-captured Store screenshots.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @c0desurfer 👋

Thanks for the contribution and fixes 💪

`Open Finder Selection` passes exact filesystem paths through `input.trim()`. A selected file named `report ` becomes `report`, which can target a different file.

Could you preserve Finder paths verbatim? One option is to encode them as file URLs before normalization:

```typescript
import { pathToFileURL } from "node:url";

await send(
  "open",
  openFields(items.map((item) => pathToFileURL(item.path).href)),
);
```

I verified this preserves the trailing space. Regression coverage would be helpful too, but it’s not a requirement for this PR.

---

I’ll convert this PR into a draft after submitting this review. Please press Ready for review when it’s ready and we’ll have a look 😊

**Comment by c0desurfer** [comment]
Fixed in 8f69017c. Finder selections are now converted to file URLs before they enter the shared path normalizer, preserving leading and trailing whitespace verbatim. I also added a regression test that round-trips a whitespace-bearing Finder path.

**Comment by c0desurfer** [comment]
Replacement PR: #31152. GitHub would not reopen this PR after its head history was repaired. The replacement keeps the same single clean extension commit and includes the requested Finder file-URL conversion plus regression coverage. @pernielsentikaer
